scripts/ASTImportTest.sh

```diff
@@ -108,33 +178,25 @@ do
     OUTPUT=$("$SPLITSOURCES" "$solfile")
     SPLITSOURCES_RC=$?
     set -e
-    if [ ${SPLITSOURCES_RC} == 0 ]
+
+    if [[ $SPLITSOURCES_RC == 0 ]]
     then
-        # echo $OUTPUT
-        NSOURCES=$((NSOURCES - 1))
-        for i in $OUTPUT;
-        do
-            testImportExportEquivalence "$i" "$OUTPUT"
-            NSOURCES=$((NSOURCES + 1))
-        done
-    elif [ ${SPLITSOURCES_RC} == 1 ]
+        IFS=' ' read -ra OUTPUT_ARRAY <<< "$OUTPUT"
+        testImportExportEquivalence "$solfile" "${OUTPUT_ARRAY[@]}"
+    elif [ $SPLITSOURCES_RC == 1 ]
     then
-        testImportExportEquivalence "$solfile"
-    elif [ ${SPLITSOURCES_RC} == 2 ]
+        testImportExportEquivalence "$solfile" "$solfile"
+    elif [ $SPLITSOURCES_RC == 2 ]
     then
         # The script will exit with return code 2, if an UnicodeDecodeError occurred.
         # This is the case if e.g. some tests are using invalid utf-8 sequences. We will ignore
         # these errors, but print the actual output of the script.
-        echo -e "\n${OUTPUT}\n"
-        testImportExportEquivalence "$solfile"
+        printError "\n\n${OUTPUT}\n\n"
+        testImportExportEquivalence "$solfile" "$solfile"
     else
         # All other return codes will be treated as critical errors. The script will exit.
-        echo -e "\nGot unexpected return code ${SPLITSOURCES_RC} from ${SPLITSOURCES}. Aborting."
-        echo -e "\n${OUTPUT}\n"
-
-        cd "$WORKINGDIR"
-        # Delete temporary files
-        rm -rf "$FILETMP"
```

Comment:
You're no longer deleting the files in case of failure. I guess it might be useful for debugging but just wanted to make sure it's an intentional change.

Reply:
Yeah, I think it could help to debug potential problems. However, I will now also output the filename of the file that caused that problem.

Reply:
I will add the deleting of the files again. As everything should be reproducible, if the user knows the solidity filename causing the problem.